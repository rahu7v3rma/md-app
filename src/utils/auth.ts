import { CommonActions } from '@react-navigation/native';
import * as Keychain from 'react-native-keychain';

import { navigationRef } from '@/navigation';
import { resetNotifications } from '@/services/notification';
import { resetStore } from '@/store';

export const setAuthToken = (token: string) => {
    Keychain.setGenericPassword('token', token);
};

export const getAuthToken = (): Promise<string | null> => {
    return Keychain.getGenericPassword().then((credentials) =>
        credentials === false ? null : credentials.password
    );
};

export const AUTHORIZATION_HEADER_NAME = 'X-Authorization';

export const getAuthorizationHeaderValue = () => {
    return getAuthToken()
        .then((authToken) => `Token ${authToken || ''}`)
        .catch(() => undefined);
};

export const resetAuthToken = () => {
    Keychain.resetGenericPassword();
};

export const logoutAction = (skipNavigation: boolean = false) => {
    resetStore();
    resetNotifications();
    resetAuthToken();

    if (!skipNavigation) {
        navigationRef.dispatch(
            CommonActions.reset({
                index: 0,
                routes: [{ name: 'SignIn' }]
            })
        );
    }
};
