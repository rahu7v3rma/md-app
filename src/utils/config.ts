import config from 'react-native-config';

export const API_BASE_URL = config.API_BASE_URL;
