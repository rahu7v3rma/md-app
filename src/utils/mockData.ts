import {
    Attachment,
    Channel,
    ChannelMemberResponse,
    StreamChat,
    UserResponse
} from 'stream-chat';

import { EditLogBlood } from '@/types/log';

export const MockData: any = {
    journey: [
        {
            is_completed: false,
            order: 1,
            block: {
                name: 'cs 101',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
                locked: false,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 1,
                        title: 'programming',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                    }
                },
                {
                    is_completed: true,
                    order: 2,
                    lesson: {
                        id: 2,
                        title: 'db',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                    }
                }
            ]
        },
        {
            is_completed: false,
            order: 2,
            block: {
                name: 'cs 102',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
                locked: false,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 1,
                        title: 'programming',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                    }
                },
                {
                    is_completed: false,
                    order: 2,
                    lesson: {
                        id: 2,
                        title: 'db',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                    }
                }
            ]
        },
        {
            is_completed: false,
            order: 3,
            block: {
                name: 'cs 103',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs102.png',
                locked: true,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 3,
                        title: 'algorithms',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/algorithms.png'
                    }
                },
                {
                    is_completed: false,
                    order: 2,
                    lesson: {
                        id: 4,
                        title: 'patterns',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/patterns.png'
                    }
                }
            ]
        }
    ],
    activeBlock: {
        is_completed: false,
        order: 1,
        block: {
            name: 'cs 101',
            icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
            locked: false,
            description:
                'Discover how to eliminate anxiety about what you put in your mouth'
        },
        user_lessons: [
            {
                is_completed: false,
                order: 1,
                lesson: {
                    id: 1,
                    title: 'programming',
                    icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                }
            },
            {
                is_completed: true,
                order: 2,
                lesson: {
                    id: 2,
                    title: 'db',
                    icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                }
            }
        ]
    },
    logTabData: [
        {
            icon: '<svg> </svg> ',
            id: 1,
            screen: 'LogBlood',
            title: 'Log Blood Glucose',
            type: 'UserGlucose'
        },
        {
            id: 2,
            icon: '<svg> </svg> ',
            title: 'Log Hydration',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 3,
            icon: '<svg> </svg> ',
            title: 'Log Hydration3',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 4,
            icon: '<svg> </svg> ',
            title: 'Log Hydration4',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 5,
            icon: '<svg> </svg> ',
            title: 'Log Hydration5',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 6,
            icon: '<svg> </svg> ',
            title: 'Log Hydration6',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 7,
            icon: '<svg> </svg> ',
            title: 'Log Hydration7',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        }
    ],
    dailyCompletedTasks: {
        UserDrink: false,
        UserExercise: false,
        UserFast: true,
        UserGlucose: true,
        UserInsulin: true,
        UserLesson: false,
        UserMedication: true,
        UserWeight: false
    },
    pickerValues: {
        hydration: {},
        weight: {},
        glucose: {},
        medication: {},
        insulin: {},
        exercise: {}
    }
};

export const mockContentReducerState = {
    loading: false,
    navigationKey: '',
    journey: [
        {
            is_completed: false,
            order: 1,
            block: {
                name: 'block1',
                icon: '/media/75beddf6-e8e1-4a0c-aaf3-7890c94f8aab.png',
                locked: false,
                description: 'description1'
            },
            user_lessons: [
                {
                    lesson: {
                        id: 1,
                        title: 'lesson1',
                        icon: '/media/b27dd75e-43fb-4200-add0-50020abe0b06.png'
                    },
                    is_completed: true,
                    order: 1
                },
                {
                    lesson: {
                        id: 2,
                        title: 'lesson2',
                        icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
                    },
                    is_completed: false,
                    order: 2
                },
                {
                    lesson: {
                        id: 3,
                        title: 'lesson3',
                        icon: '/media/90425ab2-931b-41d3-bcd0-e44ada75c60f.png'
                    },
                    is_completed: false,
                    order: 3
                }
            ]
        }
    ]
};

export const mockContentSelector = {
    loading: false,
    activeLesson: {
        lesson: {
            id: 2,
            title: 'lesson2',
            icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
        },
        is_completed: false,
        order: 2
    },
    activeBlock: {
        is_completed: false,
        order: 1,
        block: {
            name: 'block1',
            icon: '/media/75beddf6-e8e1-4a0c-aaf3-7890c94f8aab.png',
            locked: false,
            description: 'description1'
        },
        user_lessons: [
            {
                lesson: {
                    id: 1,
                    title: 'lesson1',
                    icon: '/media/b27dd75e-43fb-4200-add0-50020abe0b06.png'
                },
                is_completed: false,
                order: 1
            },
            {
                lesson: {
                    id: 2,
                    title: 'lesson2',
                    icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
                },
                is_completed: true,
                order: 2
            }
        ]
    }
};

export const logItem: EditLogBlood = {
    type: 'UserLesson',
    id: 1,
    log_time: '',
    amount: 0,
    unit: '',
    measurement_type: ''
};

export const mockAttachment: Attachment = {
    image_url:
        'https://us-east.stream-io-cdn.com/1237334/images/bbc8c8f0-fd25-48b3-8967-bd256bc5eb1b.anime.png?Key-Pair-Id=APKAIHG36VEWPDULE23Q&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly91cy1lYXN0LnN0cmVhbS1pby1jZG4uY29tLzEyMzczMzQvaW1hZ2VzL2JiYzhjOGYwLWZkMjUtNDhiMy04OTY3LWJkMjU2YmM1ZWIxYi5hbmltZS5wbmc~Km9oPTIyNCpvdz0yMjQqIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjk4NzQwNTg0fX19XX0_&Signature=Rdm3s~09cXNa5NxdRZH2RGk0~MByiEytIEt6~GzkQN2cuOrNr8OPTxj1kEFFpY8rQYIIVyJ6Edp-Fz4OyWepBqEAIYXURYnof~syP4sdQT84hsowJbxuh1fQUuQPB3MVq6lxL3cKIkRFemTrSdTXS~dr6Uxmoo7xT8~5h6hp9JN2i19UWUSac-8QBeObyHvCqLPpicx3NTkQHKrjDEifrhROhWfspbaGJJPQ52D168XvGE3YgIsZhsur0sXdooVJYl4Nc2kh9tIH35cbS9d6Y5oAQsFLDFZiRgh-C1zjDroAL-F1rtZ~25mxhnlgDyN6CJNCB4R5B4-Oxehps-nGTw__&oh=224&ow=224',
    mime_type: 'image/png',
    originalFile: {
        extension: null,
        fileSize: 14380,
        filename: 'anime.png',
        height: 225,
        playableDuration: null,
        source: 'picker',
        uri: 'ph://6F90DC2A-FCC2-4512-9B37-1BC0DB175580/L0/001',
        width: 225
    },
    original_height: 224,
    original_width: 224,
    type: 'giphy'
};
export const currentUser: UserResponse = {
    id: '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
    role: 'customer',
    created_at: '2023-02-28T21:11:55.069783Z',
    updated_at: '2023-09-04T11:19:39.641563Z',
    last_active: '2023-10-06T17:15:26.915989907Z',
    banned: false,
    online: true,
    invisible: false,
    devices: [],
    mutes: [],
    channel_mutes: [],
    unread_count: 0,
    total_unread_count: 0,
    unread_channels: 0,
    language: '',
    name: 'Test User 1',
    image: null
};

export const members: Record<string, ChannelMemberResponse> = {
    '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528': {
        user_id:
            '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
        user: {
            id: '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
            role: 'customer',
            created_at: '2023-02-28T21:11:55.069783Z',
            updated_at: '2023-09-04T11:19:39.641563Z',
            last_active: '2023-10-06T16:49:52.895391019Z',
            banned: false,
            online: true,
            name: 'Test User 1',
            image: null
        },
        created_at: '2023-03-18T18:07:35.050513Z',
        updated_at: '2023-03-18T18:07:35.050513Z',
        banned: false,
        shadow_banned: false,
        role: 'member',
        channel_role: 'channel_member'
    },
    e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9: {
        user_id:
            'e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9',
        user: {
            id: 'e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9',
            role: 'coach',
            created_at: '2023-02-28T20:53:38.035486Z',
            updated_at: '2023-09-04T11:19:47.822109Z',
            last_active: '2023-10-03T20:24:58.357617Z',
            banned: false,
            online: false,
            name: 'Coach Coach Test Coach 1',
            image: 'coach image'
        },
        created_at: '2023-03-18T18:07:35.050513Z',
        updated_at: '2023-03-18T18:07:35.050513Z',
        banned: false,
        shadow_banned: false,
        role: 'owner',
        channel_role: 'channel_member'
    }
};

export class ChannelGenerator {
    channel: Channel | null;
    _client: StreamChat | undefined;
    user_id: string;
    chatProfile_token: string;

    constructor(
        _client: StreamChat,
        user_id: string,
        chatProfile_token: string
    ) {
        this._client = _client;
        this.channel = null;
        this.user_id = user_id;
        this.chatProfile_token = chatProfile_token;
    }

    async initChannel() {
        if (this._client) {
            await this._client.connectUser(
                { id: this.user_id },
                this.chatProfile_token
            );
            let channels = await this._client.queryChannels({
                members: { $in: [this.user_id] }
            });
            this.channel = channels[0];
        }
    }

    setName(name: string) {
        if (this.channel?.data) {
            this.channel.data.name = name;
        }
    }

    setImage(image: string) {
        if (this.channel?.data) {
            this.channel.data.image = image;
        }
    }

    setMembers(_members: Record<string, ChannelMemberResponse>) {
        if (this.channel?.state) {
            this.channel.state.members = _members;
        }
    }
}
