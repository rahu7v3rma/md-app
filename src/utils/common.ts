import { once } from 'lodash';
import { PixelRatio, Platform } from 'react-native';

import { API_BASE_URL } from './config';

const warnLocalBackend = once(() =>
    console.warn('Working with a local backend')
);

export const COMMON = {
    get isIos() {
        return Platform.OS === 'ios';
    },
    get apiBaseUrl() {
        if (API_BASE_URL) {
            return API_BASE_URL;
        } else {
            warnLocalBackend();

            return this.isIos
                ? 'http://127.0.0.1:8000/'
                : 'http://10.0.2.2:8000/';
        }
    },
    responsiveSize(value: number) {
        return PixelRatio.getPixelSizeForLayoutSize(value) * 0.35;
    },
    stringFormat(s: string, ...args: any[]) {
        return s.replace(/{([0-9]+)}/g, (match, index) =>
            typeof args[index] === 'undefined' ? match : args[index]
        );
    },
    delay(ms: number) {
        return new Promise((resolve) => setTimeout(resolve, ms));
    },
    retryOperation(
        operation: () => Promise<any>,
        delayMs: number,
        retries: number
    ) {
        return new Promise<any>((resolve, reject) => {
            return operation()
                .then(resolve)
                .catch((reason) => {
                    if (retries > 0) {
                        return COMMON.delay(delayMs)
                            .then(
                                COMMON.retryOperation.bind(
                                    null,
                                    operation,
                                    delayMs,
                                    retries - 1
                                )
                            )
                            .then(resolve)
                            .catch(reject);
                    }

                    return reject(reason);
                });
        });
    }
};
