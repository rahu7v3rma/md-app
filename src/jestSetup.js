import mockRNCameraRoll from '@react-native-camera-roll/camera-roll/src/__mocks__/nativeInterface';
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock';
import mockRNDeviceInfo from 'react-native-device-info/jest/react-native-device-info-mock';

import { MockData } from './utils/mockData';

// setImmediate and clearImmediate aren't mocked by useFakeTimers if they are
// not set to a function (https://github.com/jestjs/jest/pull/11599)
global.setImmediate = jest.fn();
global.clearImmediate = jest.fn();
jest.useRealTimers();

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

jest.mock('react-native-keychain', () => {
    return {
        SECURITY_LEVEL_ANY: 'MOCK_SECURITY_LEVEL_ANY',
        SECURITY_LEVEL_SECURE_SOFTWARE: 'MOCK_SECURITY_LEVEL_SECURE_SOFTWARE',
        SECURITY_LEVEL_SECURE_HARDWARE: 'MOCK_SECURITY_LEVEL_SECURE_HARDWARE',
        setGenericPassword: jest.fn().mockResolvedValue(),
        getGenericPassword: jest.fn().mockResolvedValue(false),
        resetGenericPassword: jest.fn().mockResolvedValue()
    };
});

jest.mock('react-native-splash-screen', () => {
    return {
        hide: jest.fn()
    };
});

export let mockUnwrap = jest.fn().mockImplementation(() => {
    return Promise.resolve({});
});
export const mockDispatch = jest.fn().mockImplementation(() => {
    return {
        unwrap: mockUnwrap
    };
});
jest.mock('@/hooks', () => ({
    useAppDispatch: () => mockDispatch,
    useChannelPreviewInfo: () => mockDispatch
}));
export const mockChannelId = 'mockChannelId';
export const mockChatClient = jest.fn().mockImplementation(() => ({
    userID: 1,
    chat_id: 1,
    queryChannels: jest.fn().mockImplementation(() => {
        return Promise.resolve([
            {
                cid: 99
            }
        ]);
    }),
    activeChannels: {
        mockChannelId
    }
}));
jest.mock('@/contexts/appChat', () => ({
    useAppChat: () => {
        return {
            chatClient: mockChatClient(),
            setActiveChatChannel: jest.fn()
        };
    }
}));
export const mockRNSha256 = jest
    .fn()
    .mockImplementation(() => Promise.resolve('123'));
jest.mock('react-native-sha256', () => {
    return {
        sha256: mockRNSha256
    };
});

jest.mock('react-native-share', () => ({
    default: jest.fn()
}));

jest.mock('@/services/api', () => ({
    getImageUploadCredentials: jest.fn(() => ({
        keyPrefix: 'keyPrefix',
        fileName: 'fileName',
        bucketName: 'bucketName'
    }))
}));

jest.mock('@/reducers/content', () => ({
    ContentSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        journey: MockData.journey,
        navigationKey: 'Main.Home',
        activeBlock: MockData.activeBlock,
        activeLesson: {
            lesson: {}
        },
        journeyBlocks: MockData.journeyBlocks
    })),
    loadJourney: jest.fn(),
    getActiveLesson: jest.fn()
}));

export const mockChatProfile = jest.fn(() => ({
    apiKey: 'apiKey',
    token: 'token',
    userId: 'userId',
    firebasePushProviderName: 'firebasePushProviderName'
}));
export const mockDeleteAccount = jest.fn();
jest.mock('@/reducers/user', () => ({
    UserSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        userInfo: {
            username: 'user1',
            first_name: 'Some',
            last_name: 'Name',
            email: 'user1@example.example'
        },
        userProfile: {
            diabetes_type: 'Prediabetes',
            image: null,
            onboarding_form_url: null
        },
        coach: {
            first_name: 'name',
            profile_image: 'profile_image'
        },
        chatProfile: mockChatProfile(),
        hasCoachChat: false
    })),
    userLogin: jest.fn(),
    changePassword: jest.fn(),
    resetPassword: jest.fn(),
    resetPasswordConfirm: jest.fn(),
    resetPasswordVerify: jest.fn(),
    getCoach: jest.fn(),
    refreshProfileSession: jest.fn(),
    getProfile: jest.fn(),
    searchFilterAction: jest.fn(),
    clearSearchText: jest.fn(),
    deleteAccount: mockDeleteAccount
}));
export const mockRNFSExists = jest
    .fn()
    .mockImplementation(() => Promise.resolve(false));
export const mockRNFSCachesDirectoryPath = 'mockCachesDirectoryPath';
export const mockRNFSDownloadFile = jest.fn().mockImplementation(() => ({
    promise: Promise.resolve({ statusCode: 200 })
}));
jest.mock('react-native-fs', () => {
    return {
        readFile: jest.fn(),
        exists: mockRNFSExists,
        CachesDirectoryPath: mockRNFSCachesDirectoryPath,
        downloadFile: mockRNFSDownloadFile,
        unlink: jest.fn().mockImplementation(() => Promise.resolve())
    };
});
jest.mock('react-native-image-crop-picker', () => {
    return {
        openPicker: jest.fn(),
        launchImageLibrary: jest.fn(),
        openCamera: jest.fn()
    };
});

jest.mock('@react-native-camera-roll/camera-roll', () => mockRNCameraRoll);
jest.mock('react-native-reanimated', () => ({
    ...require('react-native-reanimated/mock')
}));
jest.mock('react-native-device-info', () => mockRNDeviceInfo);
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

jest.mock('@/reducers/track', () => ({
    TrackSelectors: jest.fn().mockImplementation(() => ({
        trackFastState: '',
        trackStartedAt: '',
        trackTimeLimitInSeconds: 57600,
        trackedTimeInSeconds: 0
    })),
    trackStart: jest.fn(),
    trackUpdate: jest.fn()
}));

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo);

jest.mock('@notifee/react-native', () =>
    require('@notifee/react-native/jest-mock')
);

export const mockOnTokenRefresh = jest.fn();
jest.mock('@react-native-firebase/messaging', () => ({
    __esModule: true,
    default: () => ({ onTokenRefresh: mockOnTokenRefresh }),
    FirebaseMessagingTypes: jest.fn()
}));

export const mockNavigate = jest.fn();
export const mockPop = jest.fn();
export const mockGoBack = jest.fn();
export const mockReplace = jest.fn();
export const mockNavigationDispatch = jest.fn();
export const mockUseRoute = jest.fn().mockImplementation(() => ({
    params: {
        id: 1,
        unit: 'kg',
        code: '11111',
        url: 'https://www.google.com/'
    }
}));

jest.mock('@/services/notification', () => ({
    resetNotifications: jest.fn(),
    registerBackgroundNotificationsListener: jest.fn(),
    getFCMToken: jest.fn().mockImplementation(() => {
        return Promise.resolve({});
    }),
    triggerFirebaseForegroundNotification: jest.fn(),
    registerFirebaseMessageListener: jest.fn(),
    requestNotificationPermissions: jest.fn()
}));

jest.mock('@react-navigation/native', () => {
    const real = jest.requireActual('@react-navigation/native');
    return {
        ...real,
        useNavigation: jest.fn().mockImplementation(() => {
            return {
                canGoBack: jest.fn().mockImplementation(() => {
                    return true;
                }),
                goBack: mockGoBack,
                navigate: mockNavigate,
                pop: mockPop,
                replace: mockReplace,
                dispatch: mockNavigationDispatch
            };
        }),
        useRoute: mockUseRoute,
        useIsFocused: jest.fn().mockImplementation(() => ({
            isFocused: true
        })),
        useFocusEffect: jest.fn()
    };
});

export const mockToast = jest.fn();

jest.mock('react-native-toast-message', () => ({
    show: mockToast
}));

export const mockGetChannelDisplayImage = jest.fn();
jest.mock('@/services/chat', () => ({
    getChannelDisplayImage: mockGetChannelDisplayImage
}));

export const mockConnectUser = jest.fn(() => Promise.resolve());
export const mockDisconnectUser = jest.fn();
export const mockGetInstance = jest.fn(() => ({
    connectUser: mockConnectUser,
    disconnectUser: mockDisconnectUser
}));
jest.mock('stream-chat', () => ({
    StreamChat: {
        getInstance: mockGetInstance
    }
}));

jest.mock('@aws-sdk/client-s3', () => ({
    S3Client: jest.fn(() => ({
        send: jest.fn(() => Promise.resolve())
    })),
    PutObjectCommand: jest.fn().mockImplementation(() => {
        return {};
    })
}));

jest.mock('react-native-pdf', () => null);
