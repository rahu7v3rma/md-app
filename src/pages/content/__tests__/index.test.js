import { configureStore, createSlice } from '@reduxjs/toolkit';
import React from 'react';
import { Provider } from 'react-redux';
import { act, create } from 'react-test-renderer';

import { mockGoBack, mockNavigate, mockUseRoute } from '@/jestSetup';
import LessonContent from '@/pages/content';
import { mockContentSelector } from '@/utils/mockData';

jest.mock('@/pages/content/components/textContent');

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: () => [false, () => {}],
    useEffect: jest.fn()
}));

jest.mock('@/reducers/content', () => ({
    ContentSelectors: jest.fn().mockImplementation(() => mockContentSelector),
    loadLesson: jest.fn(),
    completeLesson: jest.fn(),
    loadJourney: jest.fn(),
    getNavigationKey: {
        navigation: jest.fn()
    }
}));

test('snapshot', async () => {
    let tree;
    await act(() => {
        tree = create(
            <Provider
                store={configureStore({
                    reducer: createSlice({ name: 'mock' }).reducer
                })}
            >
                <LessonContent />
            </Provider>
        );
    });
    expect(tree).toMatchSnapshot();
});

test('incomplete lesson', async () => {
    mockUseRoute.mockReturnValueOnce({
        params: {
            lessonId: 1,
            lessonName: 'lesson1',
            navigationKey: 'Block1',
            type: 'Block'
        }
    });

    let tree;
    await act(() => {
        tree = create(
            <Provider
                store={configureStore({
                    reducer: createSlice({ name: 'mock' }).reducer
                })}
            >
                <LessonContent />
            </Provider>
        );
    });

    await act(() => tree.root.findByProps({ primary: true }).props.onPress());
    expect(mockNavigate).toBeCalledWith('Main.Home', {
        lessonId: 1
    });
});

test('completed lesson', async () => {
    mockUseRoute.mockReturnValueOnce({
        params: {
            lessonId: 2,
            lessonName: 'lesson2',
            navigationKey: 'Block2',
            type: 'Block'
        }
    });

    let tree;
    await act(() => {
        tree = create(
            <Provider
                store={configureStore({
                    reducer: createSlice({ name: 'mock' }).reducer
                })}
            >
                <LessonContent />
            </Provider>
        );
    });

    await act(() => tree.root.findByProps({ primary: true }).props.onPress());
    expect(mockGoBack).toBeCalled();
});
