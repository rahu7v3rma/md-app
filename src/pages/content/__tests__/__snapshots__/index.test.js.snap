// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 22,
            "paddingHorizontal": 19,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "flex-start",
              "width": "20%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 100,
                "height": 48,
                "justifyContent": "center",
                "margin": 0,
                "opacity": 1,
                "paddingHorizontal": 15.5,
                "paddingVertical": 24,
                "shadowColor": "rgba(0, 0, 0, 0.3)",
                "shadowOffset": {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "width": 48,
              }
            }
            testID="leftButton"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="none"
              focusable={false}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "height": "auto",
                    "width": "auto",
                  },
                  {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L18.7071 6.70711ZM5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L5.29289 17.2929ZM17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L17.2929 18.7071ZM6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L6.70711 5.29289ZM17.2929 5.29289L5.29289 17.2929L6.70711 18.7071L18.7071 6.70711L17.2929 5.29289ZM18.7071 17.2929L6.70711 5.29289L5.29289 6.70711L17.2929 18.7071L18.7071 17.2929Z"
                  fill={4280752721}
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "width": "60%",
            }
          }
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              [
                {
                  "fontFamily": "Poppins",
                },
                undefined,
                {
                  "color": "#72996B",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "lineHeight": 24,
                },
              ]
            }
            testID="title"
          />
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "width": "20%",
            }
          }
        />
      </View>
      <View
        style={false}
      >
        <View
          style={false}
        />
      </View>
    </View>
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "color": "#1F1E22",
            "fontFamily": "Poppins",
            "fontSize": 24,
            "fontWeight": "700",
            "lineHeight": 32,
            "marginVertical": 20,
            "opacity": 1,
            "paddingHorizontal": 20,
          },
          {
            "color": "#271A51",
            "fontSize": 24,
            "fontWeight": "700",
            "lineHeight": 32,
          },
        ]
      }
    />
    <RCTScrollView
      style={
        {
          "flex": 1,
          "paddingHorizontal": 20,
        }
      }
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 0,
          "marginTop": 12,
          "paddingVertical": 5,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#72996B",
            "borderBottomWidth": 5,
            "borderColor": "#72996B",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 39.199999999999996,
            "justifyContent": "center",
            "margin": 0,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 0,
            "width": "90%",
          }
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#FFFFFF",
                "fontSize": undefined,
                "fontWeight": "700",
                "lineHeight": undefined,
              },
            ]
          }
        >
          Complete Lesson
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
